src/outputs/io_wrapper.cpp

```diff
@@ -97,7 +97,7 @@ std::size_t IOWrapper::Read(void *buf, IOWrapperSizeT size, IOWrapperSizeT count
 //! \fn int IOWrapper::Read_all(void *buf, IOWrapperSizeT size, IOWrapperSizeT count)
 //  \brief wrapper for {MPI_File_read_all} versus {std::fread}
 
-std::size_t IOWrapper::Read_all(void *buf, IOWrapperSizeT size, IOWrapperSizeT count) {
+std::ptrdiff_t IOWrapper::Read_all(void *buf, IOWrapperSizeT size, IOWrapperSizeT count) {
```

Comment:
Can we make count and size const, here and for the below methods.

Reply:
Sure. I actually didn't realize you could mark arguments const in the definition, but not in the declaration, but that actually turns out to be a pretty reasonable thing to do. Google even recommends it! https://abseil.io/tips/109